### ISSUE TITLE
[Bug]: Calling `subscribeToInAppMessage` is still needed to display IAMs on iOS default UI

### ISSUE THREAD

**Issue by svenlombaert** [braze-inc/braze-react-native-sdk#212]
### What problem are you facing?

I'm unable to follow the Braze iOS integration guide (upgrading from 1.x appboy-sdk), specifically this part:

```objective-c
  // Setup Braze bridge
  NSURL *jsCodeLocation =
      [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index"];
  RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL:jsCodeLocation
                                                      moduleName:@"<YOUR_PROJECT_NAME>"
                                               initialProperties:nil
                                                   launchOptions:launchOptions];

  // Configure views in the application
  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];
```

react-native-navigation doesn't have the concept of 'one' Rootview. I have a feeling I have a half working integration now. I can see my user data in the braze dashboard and I do receive push notifications when trying out IAM. Once I open the IAM there's nothing displayed, I suspect this is coming because I don't have the above part?

### Workarounds

I have no idea of a workaround for now except skipping that part of the installation instructions

### Ideal Solution

Make it work with react-native-navigation

### Other Information

Currently, my set-up looks like this:
```objective-c
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  
  self.moduleName = @"Delta";
  // You can add your custom initial props in the dictionary below.
  // They will be passed down to the ViewController used by React Native.
  self.initialProps = @{};
  
  RCTAppSetupPrepareApp(application, false);

  #ifdef FB_SONARKIT_ENABLED
    InitializeFlipper(application);
  #endif

  NSURL *documentsDir = [[NSFileManager.defaultManager URLsForDirectory:NSDocumentDirectory inDomains:NSUserDomainMask] firstObject];
  NSLog(@"🚒 Documents path: %@", documentsDir.path);
  NSURL *containerDir = [[ContainerHelper new] containerURL];
  NSLog(@"🚒 Shared container path: %@", containerDir.path);

  NSURLCache.sharedURLCache = [[NSURLCache alloc] initWithMemoryCapacity: 8 * 1024 * 1024 diskCapacity: 60 * 1024 * 1024 diskPath: nil];

  // Needed for Braze
  UNUserNotificationCenter* center = [UNUserNotificationCenter currentNotificationCenter];
  center.delegate = self;

  NSString *configuration = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"CONFIGURATION"];
  NSString *firebaseConfig = [[NSBundle mainBundle] pathForResource:[NSString stringWithFormat:@"GoogleService-Info-%@", configuration] ofType:@"plist"];

  FIROptions *options = [[FIROptions alloc] initWithContentsOfFile:firebaseConfig];
  [FIRApp configureWithOptions:options];

  #if DEBUG
    NSString *brazeApiKey = @"apiKey";
  #else
    NSString *brazeApiKey = @"apiKey";
  #endif

  if (!self.bridge) {
    self.bridge = [self createBridgeWithDelegate:self launchOptions:launchOptions];
  }
  
  // Setup Braze
  BRZConfiguration *brazeConfig = [[BRZConfiguration alloc] initWithApiKey:brazeApiKey
                                                                     endpoint:@"sdk.fra-01.braze.eu"];
  // - Enable logging and customize the configuration here
  brazeConfig.logger.level = BRZLoggerLevelInfo;
  Braze *braze = [BrazeReactBridge initBraze:brazeConfig];
  
  AppDelegate.braze = braze;
  
  [ReactNativeNavigation bootstrapWithBridge:self.bridge];

  [self.window makeKeyAndVisible];

  // [REQUIRED] Register BackgroundFetch
  [[TSBackgroundFetch sharedInstance] didFinishLaunching];

  [RNSplashScreen show];

  return YES;
}
```

**Comment by svenlombaert** [comment]
Seems like I got IAM working now by adding this part on the JS side:

```typescript
Braze.subscribeToInAppMessage(true, (event) => {
      console.log('braze: ', event);
    });
```

The docs don't mention this though, they say it should go automatically.

<img width="990" alt="Screenshot 2023-05-23 at 09 31 41" src="https://github.com/braze-inc/braze-react-native-sdk/assets/2846986/1b6d20cc-b833-486b-af82-5a8e67c00760">

**Comment by jerielng** [comment]
Hi @svenlombaert, we've just released version `6.0.0` of our Braze SDK, which refines the API surface of in-app message subscriptions. This release brings our iOS layer back to the documented behavior, where you do not need to call `subscribeToInAppMessage` in addition to calling `Braze.addListener` for in-app message events (you can call either one or the other depending on your use case). Our public [documentation](https://www.braze.com/docs/developer_guide/platform_integration_guides/react_native/inapp_messages/) on braze.com will be updated over the next day or two to reflect this.

We have also refined our logic to prevent overriding any custom behavior you may have defined in the native iOS layer.

You can refer to our updated [sample app](https://github.com/braze-inc/braze-react-native-sdk/tree/master/BrazeProject) for an example integration.

Please let us know if you have any questions. Thank you!
